### PR TITLE
Fix the issue about failed to reboot guest for nmi_test

### PR DESCRIPTION
**Pull request by cliping**
Signed-off-by: Liping Cheng <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 guest_kernel_debugging.nmi_test.positive_testing.send_nmi --vt-connect-uri qemu:///system
/root/.libvirt-ci-venv-ci-runtest-wk3BTh/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:292: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 55c690304901c9181c98cad06a48825ddd30dadf
JOB LOG    : /root/avocado/job-results/job-2020-08-09T23.43-55c6903/job.log
 (1/1) type_specific.io-github-autotest-libvirt.guest_kernel_debugging.nmi_test.positive_testing.send_nmi: PASS (128.16 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 130.72 s

File: libvirt/tests/src/guest_kernel_debugging/nmi_test.py [hunk 1]
```diff
@@ -47,46 +73,53 @@ def run(test, params, env):
...
-
-        # check Non-maskable interrupts times
-        if real_nmi_times != expected_nmi_times:
-            test.fail("NMI times aren't expected %s:%s"
-                      % (real_nmi_times, expected_nmi_times))
+                try:
+                    # try to install kexec-tools on fedoraX/rhelx.y guest
+                    run_cmd_in_guest(vm, "yum install -y kexec-tools", test)
+                except Exception:
+                    test.cancel("Requires kexec-tools(or the "
```
Comment (chunfuwen) [inline libvirt/tests/src/guest_kernel_debugging/nmi_test.py:89]: change test.cancel to test.error ?

File: libvirt/tests/src/guest_kernel_debugging/nmi_test.py [hunk 2]
```diff
@@ -22,6 +22,31 @@ def run_cmd_in_guest(vm, cmd, test):
...
 
 
+def update_boot_option_and_reboot(vm, kernel_params, test):
+    """
+    Update boot option in the guest and then reboot guest
+    :params vm: vm object
+    :params cmd: a command needs to be ran
+    :params test: test object
+    """
+    session = None
```
Comment (chunfuwen) [inline libvirt/tests/src/guest_kernel_debugging/nmi_test.py:32]: no need this line

File: libvirt/tests/src/guest_kernel_debugging/nmi_test.py (hunk 1 above)
Comment (cliping) [reply libvirt/tests/src/guest_kernel_debugging/nmi_test.py]: Updated

File: libvirt/tests/src/guest_kernel_debugging/nmi_test.py (hunk 2 above)
Comment (cliping) [reply libvirt/tests/src/guest_kernel_debugging/nmi_test.py]: Updated

File: libvirt/tests/src/guest_kernel_debugging/nmi_test.py [hunk 3]
```diff
@@ -22,6 +22,30 @@ def run_cmd_in_guest(vm, cmd, test):
...
 
 
+def update_boot_option_and_reboot(vm, kernel_params, test):
+    """
+    Update boot option in the guest and then reboot guest
+    :params vm: vm object
+    :params cmd: a command needs to be ran
+    :params test: test object
+    """
+    session = vm.wait_for_login()
```
Comment (chunfuwen) [inline libvirt/tests/src/guest_kernel_debugging/nmi_test.py:32]: move into try ...catch block?

Comment (cliping) [reply libvirt/tests/src/guest_kernel_debugging/nmi_test.py]: Updated.

**Review by Yingshun — APPROVED** [review]
LGTM.

**Pull request merged**